pgtransaction/atomic.py

```diff
@@ -19,32 +18,36 @@ def __init__(
         retry,
     ):
         super().__init__(using, savepoint, durable)
-        self.connection = get_connection(self.using)
+        self.connection = transaction.get_connection(self.using)
         self.isolation_level = isolation_level
         self.retry = retry
         self._used_as_context_manager = True
         self._validate()
 
     def _validate(self):
-        if self.connection.vendor != "postgresql":  # pragma: no cover
-            raise PGAtomicConfigurationError(
-                f"pgtransaction.atomic cannot be used with {self.connection.vendor}"
-            )
-        if self.isolation_level and self.isolation_level.upper() not in (
-            "READ COMMITTED",
-            "REPEATABLE READ",
-            "SERIALIZABLE",
-        ):  # pragma: no cover
-            raise PGAtomicConfigurationError(
-                f"Isolation level {self.isolation_level} not recognised"
-            )
-        if self.isolation_level and self.connection.in_atomic_block:
-            raise PGAtomicConfigurationError(
-                "Setting the isolation level inside in a nested atomic "
-                "transaction is not permitted. Nested atomic transactions "
-                "inherit the isolation level from their parent transaction "
-                "automatically."
-            )
+        if self.isolation_level:
```

Comment:
This should probably be `if self.isolation_level or self.retry`, since the retry policy only works currently for psycopg2 errors.

Reply:
I'd argue against checking that `retry` is set. Although we currently only configure it to only catch psycopg2 errors, the code will still work if you switch out the DB as long as psycopg2 is installed. It just won't actually retry anything.

For example, say that a user uses sqlite for their test suite and only uses `retry` and never `isolation_level`. Although this might be a rare edge case, I like to avoid doing things that might be overly restrictive.

Reply:
Fair point about the tests. I suppose if we make it clear we only catch postgres errors for now then it should be safe enough.